examples/demo_basic_single_particle.jl

```diff
@@ -0,0 +1,86 @@
+# Some basic single particle cases.
+# More theoretical details can be found in F.F.Chen's Introduction to Plasma Physics and Controlled Fusion.
+
+using TestParticle
+using TestParticle: get_gc
+using TestParticleMakie
+using OrdinaryDiffEq
+using StaticArrays
+using LinearAlgebra
+using GLMakie
+
+function uniform_B(x)
+    return SA[0, 0, 1e-8]
+end
+
+function uniform_E(x)
+    return SA[1e-9, 0, 0]
+end
+
+function zero_E(x)
+    return SA[0, 0, 0]
+end
+
+function grad_B(x)
+    return SA[0, 0, 1e-8+1e-9 *x[2]]
+end
+
+function curved_B(x)
+    # satisify ∇⋅B=0
+    θ = atan(x[3]/(x[1]+3))
+    return SA[-1e-8*sin(θ), 0, 1e-8*cos(θ)]
+end
+
+function nonuniform_E(x)
+    return SA[1e-9*cos(2*π*x[1]), 0, 0]
+end
+
+function time_varying_E(x, t)
+    # return SA[0, 1e-9*cos(0.1*t), 0]
+    return SA[0, 1e-9*0.1*t, 0]
+end
+
+x0 = [1.0, 0, 0]
+v0 = [0.0, 1.0, 0.1]
+stateinit = [x0..., v0...]
+tspan = (0, 20)
+
+# E×B drift
```

Comment:
I would suggest separate each drift into a stand-alone script, since Makie seems to have limited support on multi-window display and we want to check each drift one-by-one.

Also, for each of the drift here there is an analytical solution. We may add the analytical guiding center solutions with dashed lines.